makefile

```diff
@@ -1,113 +1,177 @@
-goal:   trove.x
+################################################################################
+## USER OPTIONS
+################################################################################
 
-tarball:
-	tar cf trove.tar makefile *.f90
-        
-checkin:
-	ci -l Makefile *.f90
+EXE=trove
+pot_user = pot_H2O_Conway
 
+################################################################################
+## COMPILER OPTIONS
+################################################################################
+
+COMPILER ?= intel
+MODE ?= release
+
+# Intel
+#######
+ifeq ($(strip $(COMPILER)),intel)
+	FOR = ifort
+	FFLAGS = -cpp -qopenmp -module $(OBJDIR)
+	LAPACK = -mkl
+
+	ifeq ($(strip $(MODE)),debug)
+		FFLAGS += -O0 -g
+	else
+		FFLAGS += -O3
+	endif
+
+	# Alternative flags:
+	#FFLAGS = -ip -align -ansi-alias -traceback -qopenmp -mcmodel=medium -parallel -nostandard-realloc-lhs -module $(OBJDIR)
+	#LAPACK = -mkl=parallel
+endif
+
+# gfortran
+##########
+ifeq ($(strip $(COMPILER)),gfortran)
+	FOR = gfortran
+	#FFLAGS = -cpp -std=gnu -fbacktrace -fopenmp -march=native -ffree-line-length-512 -fallow-argument-mismatch -fcray-pointer -I$(OBJDIR) -J$(OBJDIR)
+	FFLAGS = -cpp -std=gnu -fbacktrace -fopenmp -march=native -ffree-line-length-512 -fcray-pointer -I$(OBJDIR) -J$(OBJDIR)
+
+	ifeq ($(strip $(MODE)),debug)
+		FFLAGS += -O0 -g -Wunderflow
+	else
+		FFLAGS += -O3
+	endif
+
+	# Use non-MKL LAPACK:
+	#LAPACK = -llapack -lblas
+
+	# Use MKL LAPACK:
+	LAPACK += -L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
+endif
```

Comment:
This clashes with having a default compiler option. In my opinion it would be best to set a sensible default based on the most commonly used compilers (which is intel for Sergey; not sure about German collaborators).

Reply:
I meant more in case someone accidentally sets `COMPILER` to `intek` or `gfortrn` (or tries to use a different one). Catching this here would make it clearer than having undefined variables later.

Reply:
Yep, you're right. Added error when undefined compiler specified.